packages/tenant-process/src/services/tenantService.ts

```diff
@@ -381,6 +385,70 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async addDeclaredAttribute({
+      tenantAttributeSeed,
+      authData,
+      correlationId,
+    }: {
+      tenantAttributeSeed: ApiDeclaredTenantAttributeSeed;
+      authData: AuthData;
+      correlationId: string;
+    }): Promise<Tenant> {
+      logger.info(
+        `Add declared attribute ${tenantAttributeSeed.id} to requester tenant ${authData.organizationId}`
+      );
+      const targetTenant = await retrieveTenant(
+        authData.organizationId,
+        readModelService
+      );
+
+      const maybeDeclaredTenantAttribute = targetTenant.data.attributes.find(
+        (attr) => attr.id === tenantAttributeSeed.id
+      );
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...targetTenant.data,
+        updatedAt: new Date(),
+      };
+      if (!maybeDeclaredTenantAttribute) {
+        // assigning attribute for the first time
+        updatedTenant = {
+          ...updatedTenant,
+          attributes: [
+            ...targetTenant.data.attributes,
+            {
+              id: unsafeBrandId(tenantAttributeSeed.id),
+              type: tenantAttributeType.DECLARED,
+              assignmentTimestamp: new Date(),
+              revocationTimestamp: undefined,
+            },
+          ],
+        };
+      } else {
+        if (
+          maybeDeclaredTenantAttribute.type !== "PersistentDeclaredAttribute"
+        ) {
+          throw attributeNotFound(maybeDeclaredTenantAttribute.id);
+        }
+        // re-assigning attribute if it was revoked
+        updatedTenant = updateAttribute({
+          updatedTenant,
+          targetTenant,
+          attributeId: unsafeBrandId(tenantAttributeSeed.id),
+        });
+      }
+      const event = toCreateEventTenantDeclaredAttributeAssigned(
+        targetTenant.data.id,
+        targetTenant.metadata.version,
+        updatedTenant,
+        unsafeBrandId(tenantAttributeSeed.id),
+        correlationId
+      );
+      await repository.createEvent(event);
+      return updatedTenant;
+    },
+
```

Comment:
Overall, what's the desired behavior in case we are assigning an attribute that is already assigned (and not revoked)? Should we throw an error like `attributeAlreadyAssigned`?